src/lib.rs

```diff
@@ -534,4 +534,123 @@ impl<'a, 'b> Builder<'a, 'b> {
 
         util::create_helper(dir, self.prefix, self.suffix, self.random_len, dir::create)
     }
+
+    /// Attempts to create a temporary file (or file-like object) using the
+    /// provided closure inside of [`std::env::temp_dir()`]. The closure is
+    /// passed a temporary file path and returns an [`std::io::Result`]. If the
+    /// closure returns one of the following errors, then another randomized
+    /// file path is tried:
+    ///  - [`std::io::ErrorKind::AlreadyExists`]
+    ///  - [`std::io::ErrorKind::AddrInUse`]
+    ///
+    /// This can be helpful for taking full control over the file creation, but
+    /// leaving the temporary file path construction up to the library. This
+    /// also enables creating a temporary UNIX domain socket, since it is not
+    /// possible to bind to a socket that already exists.
+    ///
+    /// Note that [`Builder::append`] is ignored when using [`Builder::make`].
+    ///
+    /// # Security
+    ///
+    /// See [the security][security] docs on `NamedTempFile`.
```

Comment:
Also note that this function will not:

- Ensure a safe mode.
- That the file doesn't already exist.

Both of these are responsibilities of closure.

Reply:
And below.

Reply:
Good call. I added a bunch of docs for the safety considerations and responsibilities of the caller.